File: ReviewBasisOfAndroid/MVP/app/src/main/java/com/sieunguoimay/vuduydu/mvp_pattern_loginapp/animations/CanvasCircle.java
package com.sieunguoimay.vuduydu.mvp_pattern_loginapp.animations;

import android.graphics.Canvas;
import android.graphics.Paint;
import android.graphics.Point;

public class CanvasCircle {
    protected int radius;
    protected Point pos;
    protected Paint paint;

    public CanvasCircle(int x, int y, int radius){
        this.pos = new Point(x,y);
        this.radius = radius;
        this.paint = new Paint();
    }

    public void setPos(int x, int y){this.pos.set(x,y);}
    public void setRadius(int radius){this.radius =radius;}
    public Paint getPaint(){return paint;}
    public void draw(Canvas canvas){
        canvas.drawCircle(pos.x, pos.y, radius,paint);
    }

    public int getRadius() {
        return radius;
    }

}

File: Week2_LookAtCompanyProject/MusicPlayerWithMVP/YAMP/app/src/main/java/com/sieunguoimay/vuduydu/yamp/widgets/particle_system/CanvasCircle.kt
package com.sieunguoimay.vuduydu.mvp_pattern_loginapp.animations

import android.graphics.Canvas
import android.graphics.Paint
import com.sieunguoimay.vuduydu.yamp.widgets.particle_system.Vec2

open class CanvasCircle(var x: Float, var y: Float, var radius: Float) {

    var pos: Vec2

    var paint: Paint
        protected set

    init {
        this.pos = Vec2(x, y)
        this.paint = Paint()
    }

    fun setPos(x: Float, y: Float) {
        this.pos.set(x, y)
    }

    fun draw(canvas: Canvas) {
        canvas.drawCircle(pos.x.toFloat(), pos.y.toFloat(), radius.toFloat(), paint)
    }

}
